File: Trabajos previos/Sesión_4/herenciaMultiple.cpp
#include <iostream>
using namespace std;

// Se crea clase Mammal.
class Mammal {
    public:
        Mammal() {
          cout << "Mammals can give direct birth." << endl;
    }
};

// Se crea clase WingedAnimal.
class WingedAnimal {
    public:
        WingedAnimal() {
          cout << "Winged animal can flap." << endl;
    }
};

// Se crea clase Bat que es hija de forma pública ambas Mammal y WingedAnimal.
class Bat: public Mammal, public WingedAnimal {};

int main() {
    Bat b1; // Se genera el constructor de ambas clases en el mismo objeto tipo Bat.
    return 0;
}

File: Trabajos previos/Sesión_12/herenciaMultiple.py
# Se crea la primera clase madre con un método.
class Mammal:
    def mammal_info(self):
        print("Mammals can give direct birth.")

# Se crea la segunda clase madre con un método.
class WingedAnimal:
    def winged_animal_info(self):
        print("Winged animals can flap.")

# Se crea una clase Bat que hereda de ambas clases madres.
class Bat(Mammal, WingedAnimal):
    pass

# Se crea un objeto de clase Bat.
b1 = Bat()

# Se puede llamar ambos métodos bases desde la clase hija.
b1.mammal_info()
b1.winged_animal_info()